library/cartridge_check_cluster_issues.py

```diff
@@ -0,0 +1,58 @@
+#!/usr/bin/env python
+
+from ansible.module_utils.helpers import Helpers as helpers
+
+argument_spec = {
+    'allow_warnings': {'required': True, 'type': 'bool'},
+    'console_sock': {'required': True, 'type': 'str'},
+}
+
+
+def get_messages(**issues_by_level):
+    messages = []
+    for level, issues in issues_by_level.items():
+        if not issues:
+            continue
+
+        if messages:
+            messages += ['']
+
+        messages += ['Issues (%s):' % level]
+        messages += [issue['message'] for issue in issues]
+
+    return messages
+
+
+def check_cluster_issues(params):
+    allow_warnings = params['allow_warnings']
+    console_sock = params['console_sock']
+
+    control_console = helpers.get_control_console(console_sock)
+
+    issues, err = control_console.eval_res_err('''
+        return require('cartridge.issues').list_on_cluster()
+    ''')
+
+    if err:
+        helpers.warn("Received error on getting list of cluster issues: %s" % err)
```

Comment:
You forgot about return, filter(issues) will be broken

Reply:
No, this function returns a list of issues and the first error received on getting issues from other instances. E.g. if one instance isn't alive there are some replication issues and errors that this instance is unavailable.

Reply:
Let's add check for nil, err